tests/lib/rules/no-identical-names.js

```diff
@@ -66,7 +66,38 @@ ruleTester.run("no-identical-title", rule, {
             parserOptions: {
                 ecmaVersion: 6
             }
-        }
+        },
+
+        // Nested modules
+        outdent`
+            module("module1", function() {
+                module("submodule1", function() {});
+            });
+            module("module2", function() {
+                module("submodule1", function() {});
+            });
+        `,
+        outdent`
+            module("module1", function() {
+                module("submodule1", function() {
+                    module("subsubmodule1", function() {
+                        test("it1", function() {})
+                    });
+                });
+            });
+            module("module2", function() {
+                module("submodule1", function() {
+                    module("subsubmodule1", function() {
+                        test("it1", function() {})
+                    });
+                });
+            });
+        `,
+        outdent`
+            module("name1", function() {
+                module("name1", function() {});
```

Comment:
I'm not sure this should be a valid case. We definitely want to support submodules with the same name under nested modules because there's no chance for ambiguity when you look at the nested module stacks. But I'm not sure that is the case for a nested submodule with the same name as an ancestor module.

What do you think?

Reply:
Yeah I'd be open to flagging this case as a violation. But we should do that in a separate PR. My current PR is about fixing false positives whereas this change would be fixing a false negative.

Reply:
Sounds good, but would you mind removing this new "valid" test case from this PR in that case? That's just in case I lose track of things and cut a release after merging this PR but before we fix the issue I've raised.

Reply:
Sure! I removed this test case to avoid confusion.